modules/integrations/arrow-integrations-retrofit-adapter/src/main/kotlin/arrow.integrations.retrofit.adapter/ResponseCallback.kt

```diff
@@ -0,0 +1,28 @@
+package arrow.integrations.retrofit.adapter
+
+import arrow.core.Either
+import arrow.core.left
+import arrow.core.right
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.HttpException
+import retrofit2.Response
+
+class ResponseCallback<R>(private val proc: (Either<Throwable, R>) -> Unit) : Callback<R> {
+  override fun onResponse(call: Call<R>, response: Response<R>) {
```

Comment:
Same as mentioned above, I wouldn't unpack the `Response<R>` and would let users decide instead.

Reply:
You're right, I am imposing an semantics on the client and we should let then decide between `Response<R>` and just `R`. But there's no need to create an extension function for this, or to change the API. The API could be the same, we just need the to read the ReturnType at the Async2CallAdapterFactory and then create the right adapter for each response. If one uses CallK<R> him/she gets the current solution, but if he/she uuses CallK<Response<R>> him/she gets yours. WDYT?

Reply:
That's pretty much how the RxJava2 plugin works

Reply:
My concern is with discoverability, as it won't be immediately apparent that `CallK<Response<X>>` has different semantics than any other value. I'd rather it was a different function of the `CallK` API, like `asyncResponse` for example.